### PR TITLE
Update for new API and add fallback

### PR DESCRIPTION
**Pull request by Rick-Kirkham**

### CODE REVIEW

File: src/Office-Add-in-ASPNET-SSO/Office-Add-in-ASPNET-SSOManifest/Office-Add-in-ASPNET-SSO.xml [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+						<!-- Description of the Getting Started callout. resid points to a LongString resource -->
+						<Description resid="Contoso.GetStarted.Description"/>
+
+						<!-- Points to a URL resource which details how the add-in should be used. -->
+						<LearnMoreUrl resid="Contoso.GetStarted.LearnMoreUrl"/>
+					</GetStarted>
+					<!-- Function file is a HTML page that includes the JavaScript where functions for ExecuteAction will be called. 
+            Think of the FunctionFile as the code behind ExecuteFunction. -->
+					<FunctionFile resid="Contoso.DesktopFunctionFile.Url" />
```
Comment (codexeon) [inline src/Office-Add-in-ASPNET-SSO/Office-Add-in-ASPNET-SSOManifest/Office-Add-in-ASPNET-SSO.xml]: I don't think this sample needs FunctionFile, this server does not implement custom functions.

File: src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    } else {
+        // Something went wrong with authentication or the authorization of the web application.
+        loginDialog.close();
+        showResult(["Unable to successfully authenticate user or authorize application. Error is: " + message.error]);
+    }
+}
+
+// Use the Office dialog API to open a pop-up and display the sign-in page for the identity provider.
+function showLoginPopup(url) {
+	var fullUrl = location.protocol + '//' + location.hostname + (location.port ? ':' + location.port : '') + url;
```
Comment (codexeon) [inline src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js:198]: location.origin + url

File: src/Office-Add-in-ASPNET-SSO-WebAPI/Controllers/ValuesController.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            try
+            {
+                authResult = await result.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                return HttpErrorHelper.SendErrorToClient(HttpStatusCode.BadRequest, e, null);
+            }
+
+            return await GraphApiHelper.GetOneDriveFileNames(authResult.AccessToken);
```
Comment (codexeon) [inline src/Office-Add-in-ASPNET-SSO-WebAPI/Controllers/ValuesController.cs:90]: This is different than the node.js sample where we return the graph token to the browser.

Comment (codexeon) [reply src/Office-Add-in-ASPNET-SSO-WebAPI/Controllers/ValuesController.cs:90]: if we want to do it this way to show the call being made from server that is fine though.

File: src/Office-Add-in-ASPNET-SSO/Office-Add-in-ASPNET-SSOManifest/Office-Add-in-ASPNET-SSO.xml (hunk 1 above)
Comment (Rick-Kirkham) [reply src/Office-Add-in-ASPNET-SSO/Office-Add-in-ASPNET-SSOManifest/Office-Add-in-ASPNET-SSO.xml]: removed

File: src/Office-Add-in-ASPNET-SSO-WebAPI/Controllers/ValuesController.cs (hunk 3 above)
Comment (Rick-Kirkham) [reply src/Office-Add-in-ASPNET-SSO-WebAPI/Controllers/ValuesController.cs:90]: This is the way all MSAL.NET samples work. They do the token acquisition and the call to Graph in the same controller.

File: src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js (hunk 2 above)
Comment (Rick-Kirkham) [reply src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js:198]: Unfortunately, location.origin doesn't work on IE. See https://stackoverflow.com/questions/22564167/window-location-origin-gives-wrong-value-when-using-ie

File: src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/Document.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+        let filenames = [];
+        let i;
+        for (i = 0; i < result.length; i++) {
+            var innerArray = [];
+            innerArray.push(result[i]);
+            filenames.push(innerArray);
+        }
+
+        const rangeAddress = `B5:B${5 + (result.length-1)}`;
```
Comment (codexeon) [inline src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/Document.js]: This syntax is not valid for IE

Comment (Rick-Kirkham) [reply src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/Document.js]: Good catch. Fixed.

File: src/Office-Add-in-ASPNET-SSO-WebAPI/App_Start/Startup.Auth.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+namespace Office_Add_in_ASPNET_SSO_WebAPI
+{
+	public partial class Startup
+	{
+		public void ConfigureAuth(IAppBuilder app)
+		{
+            TokenValidationParameters tvps = new TokenValidationParameters
+            {
+				ValidAudience = ConfigurationManager.AppSettings["ida:Audience"],
+				ValidIssuer = ConfigurationManager.AppSettings["ida:Issuer"],
```
Comment (codexeon) [inline src/Office-Add-in-ASPNET-SSO-WebAPI/App_Start/Startup.Auth.cs]: This line makes the add-in only valid for a single Tenant, which is not the case with node.js sample. Maybe we can just say validate the issuer is Microsoft?
				IssuerValidator = (issuer, token, tvp) =>
				{
					if (issuer.StartsWith("https://login.microsoftonline.com/"))
						return issuer;
					else
						throw new SecurityTokenInvalidIssuerException("Invalid issuer");
				},

File: src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	if (exceptionMessage) {
+
+        // On rare occasions the bootstrap token is unexpired when Office validates it,
+        // but expires by the time it is sent to AAD for exchange. AAD will respond
+        // with "The provided value for the 'assertion' is not valid. The assertion has expired."
+        // Retry the call of getAccessToken (no more than once). This time Office will return a 
+        // new unexpired bootstrap token.
+        if ((exceptionMessage.indexOf("The provided value for the 'assertion' is not valid. The assertion has expired.") !== -1)
```
Comment (codexeon) [inline src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js]: Do we need to check the exact string? This will certainly break in the future.

Comment (Rick-Kirkham) [reply src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js]: Switched to the number AADSTS500133

**Pull request merged**